src/mlpack/tests/sgd_test.cpp

```diff
@@ -39,6 +41,23 @@ BOOST_AUTO_TEST_CASE(SimpleSGDTestFunction)
   BOOST_REQUIRE_SMALL(coordinates[2], 1e-7);
 }
 
```

Comment:
Might be a good idea to test the policy independently from the optimizer, we could simply test if the gradient is clipped correctly.